### DISCUSSION TITLE
How to implement style isolation when using preview mode within the code block?

### DISCUSSION THREAD

**Discussion by WinmezzZ** [web-infra-dev/rspress#764] [Q&A]
代码块的预览样式，被文档的全局样式污染了，有办法避免吗

**Comment by 10Derozan** [comment]
Which previewMode, 'internal' or 'iframe'?

**Comment by WinmezzZ** [comment]
默认吧，因为没有单独设置这个配置

mdx中的代码块默认启用了预览+展开源代码的方式，不过正好是我想要的，但是预览的样式全都被影响了，比如这个按钮


![CleanShot 2024-03-14 at 10 37 27@2x](https://github.com/web-infra-dev/rspress/assets/40154766/6d160ee8-a534-48f5-8ec4-c058a6245df2)

**Comment by 10Derozan** [comment]
Later we will support shadow dom, do you have any ideas to do it?

**Reply by WinmezzZ** [reply]
好想法，从长远角度来说是一个非常可靠的做法，但是如果从简单处理的话，我觉得只需要将文档的所有全局样式针对代码预览这个区域加个选择器过滤就行，比如

```css
button:not(.rspress-preview-card) {
  //...
}

*:not(.rspress-preview-card) {
  //...
}
```

![CleanShot 2024-03-19 at 11 23 25@2x](https://github.com/web-infra-dev/rspress/assets/40154766/a3a1d187-aadf-4585-90a9-d4d29f7d1edb)

**Reply by 10Derozan** [reply]
If the change is small and not breaking, welcome PR.
